### PR TITLE
[SERVICES-2610] Update transaction generation for Wrap SC

### PR DESCRIPTION
**Pull request by mad2sm0key**
## Reasoning
- use `SmartContractTransactionsFactory` for generating transactions for Wrap SC
- missing sender in generated transaction causes a warning during unit test runs
  
## Proposed Changes
- add generic method for creating Transaction factories for each type of SC to mx.proxy service
- update wrap transactions service methods (add sender + use factory method from mxproxy)
- fix unit tests

## How to test
- N/A

### CODE REVIEW

**Review by claudiulataretu — DISMISSED** [review]

File: src/modules/common/transaction.options.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { TokenTransfer } from '@multiversx/sdk-core/out';
```
Comment (bogdan-rosianu) [inline src/modules/common/transaction.options.ts:1]: maybe drop `/out` ? I've seen this change in Claudiu's PR

File: src/modules/wrapping/services/wrap.transactions.service.ts [hunk 2]
```diff
@@ -15,36 +15,44 @@ export class WrapTransactionsService {
...
     async unwrapEgld(
         sender: string,
         amount: string,
     ): Promise<TransactionModel> {
-        const shardID = await this.mxProxy.getAddressShardID(sender);
-        const contract = await this.mxProxy.getWrapSmartContract(shardID);
-
-        const wrappedEgldToken = await this.wrapService.wrappedEgldToken();
+        const [shardID, wrappedEgldToken] = await Promise.all([
+            this.mxProxy.getAddressShardID(sender),
```
Comment (bogdan-rosianu) [inline src/modules/wrapping/services/wrap.transactions.service.ts:36]: by looking into this implementation:
```
async getAddressShardID(address: string): Promise<number> {
        const response = await this.getService().doGetGeneric(
            `address/${address}/shard`,
        );
        return response.shardID;
    }
```
there's no need for an API call for fetching the shard ID of an address. you can use `AddressComputer` from sdk-js-core:
```
const addressComputer = new AddressComputer();
let shard = addressComputer.getShardOfAddress(address);
```

File: src/services/multiversx-communication/mx.proxy.service.ts [hunk 3]
```diff
@@ -281,4 +347,24 @@ export class MXProxyService {
         MXProxyService.smartContracts[key] = newSC;
         return newSC;
     }
+
+    private async createSmartContractTransactionsFactory(
+        contractAbiPath: string,
+        contractInterface: string,
+        chainID: string,
+    ): Promise<SmartContractTransactionsFactory> {
+        const jsonContent: string = await promises.readFile(contractAbiPath, {
```
Comment (bogdan-rosianu) [inline src/services/multiversx-communication/mx.proxy.service.ts:356]: maybe try to read from `smartContractTransactionFactories` before recomputing it?

File: src/modules/common/transaction.options.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+export class TransactionOptions {
```
Comment (bogdan-rosianu) [inline src/modules/common/transaction.options.ts:3]: I think that a better naming would be `TransactionIntent`

File: src/modules/wrapping/services/wrap.transactions.service.ts (hunk 2 above)
Comment (mad2sm0key) [reply src/modules/wrapping/services/wrap.transactions.service.ts:36]: This is a good point, but I think it's outside the scope of this PR. I would create a separate PR for this, and update all the places where we use `getAddressShardID()`

File: src/services/multiversx-communication/mx.proxy.service.ts (hunk 3 above)
Comment (mad2sm0key) [reply src/services/multiversx-communication/mx.proxy.service.ts:356]: `createSmartContractTransactionsFactory` is a private method that only creates a `SmartContractTransactionsFactory` from an abi. A check if performed in the `getSmartContractTransactionFactory` method before calling this.

File: src/modules/common/transaction.options.ts (hunk 1 above)
Comment (mad2sm0key) [reply src/modules/common/transaction.options.ts]: done

File: src/modules/common/transaction.options.ts (hunk 4 above)
Comment (mad2sm0key) [reply src/modules/common/transaction.options.ts:3]: I would rename it in a separate PR once all the other PRs upstream get merged

**Review by bogdan-rosianu — APPROVED** [review]

**Review by claudiulataretu — DISMISSED** [review]

**Review by claudiulataretu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**